### PR TITLE
docs: add `fetch` script to package.json conventions

### PR DESCRIPTION
**Pull request by snitin315**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)
- As discussed in https://github.com/eslint/eslint.org/pull/461#discussion_r1290339543 
- Added `fetch:*` scripts to package.json conventions.
- Added package.json convention to the navigation.

  Screenshot 2023-08-12 at 7 11 37 AM

#### Is there anything you'd like reviewers to focus on?
No

### CODE REVIEW

File: docs/src/contribute/package-json-conventions.md [hunk 1]
```diff
@@ -35,6 +40,12 @@ Scripts that generate a set of files from source code and / or data MUST have na
 
 If a package contains any `build:*` scripts, there MAY be a script named `build`. If so, SHOULD produce the same output as running each of the `build` scripts individually. It MUST produce a subset of the output from running those scripts.
 
+### Fetch
+
+Scripts that generate a set of files from external data or resources MUST have names that begin with `fetch`.
+
+If a package contains any `fetch:*` scripts, there MAY be a script named `fetch`. If so, SHOULD produce the same output as running each of the `build` scripts individually. It MUST produce a subset of the output from running those scripts.
```
Comment (amareshsm) [inline docs/src/contribute/package-json-conventions.md:47]:
```suggestion
If a package contains any `fetch:*` scripts, there MAY be a script named `fetch`. If so, it SHOULD produce the same output as running each of the `build` scripts individually. It MUST produce a subset of the output from running those scripts.
```

File: docs/src/contribute/package-json-conventions.md [hunk 2]
```diff
@@ -35,6 +40,12 @@ Scripts that generate a set of files from source code and / or data MUST have na
 
 If a package contains any `build:*` scripts, there MAY be a script named `build`. If so, SHOULD produce the same output as running each of the `build` scripts individually. It MUST produce a subset of the output from running those scripts.
 
+### Fetch
+
+Scripts that generate a set of files from external data or resources MUST have names that begin with `fetch`.
+
+If a package contains any `fetch:*` scripts, there MAY be a script named `fetch`. If so, SHOULD produce the same output as running each of the `fetch` scripts individually. It MUST produce a subset of the output from running those scripts.
```
Comment (snitin315) [inline docs/src/contribute/package-json-conventions.md:47]:
```suggestion
If a package contains any `fetch:*` scripts, there MAY be a script named `fetch`. If so, it SHOULD produce the same output as running each of the `fetch` scripts individually. It MUST produce a subset of the output from running those scripts.
```

**Comment by mdjermanovic** [comment]
Should we also add `"fetch"` here?

https://github.com/eslint/eslint/blob/ee2f718188d32e9888b1932fe6b9bd2a62c529a4/docs/src/contribute/package-json-conventions.md?plain=1#L17

**Comment by snitin315** [comment]
Yes, we should. I'll update the PR.

**Comment by snitin315** [comment]
@mdjermanovic updated.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Review by nzakas — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**